### PR TITLE
DOC: Improvements and additions to the matrix_normal doc

### PR DESCRIPTION
**Pull request by drpeteb**
For #5478
- Documented the relationship with the multivariate normal distribution.
- Some small clarifications made and typos fixed.

### CODE REVIEW

**Comment by drpeteb** [comment]
Dammit. I thought I'd rebased those first three commits away. Is this messy history a problem?

File: scipy/stats/_multivariate.py [hunk 1]
```diff
@@ -700,22 +700,32 @@ class matrix_normal_gen(multi_rv_generic):
...
 
+    The `allow_singular` behaviour of the `multivariate_normal`
+    distribution is not currently supported. Covariance matrices must be
+    full rank.
+
+    The `matrix_normal` distribution is closely related to the
+    `multivariate_normal` distribution. Specifically, :math:`\mathrm{Vec}(X)`
+    (the vectorisation of :math:`X`) has a multivariate normal distribution
+    with mean :math:`\mathrm{Vec}(M)` and covariance :math:`V \otimes U`
+    (where :math:`\otimes` is the Kronecker product).
```
Comment (ev-br) [inline scipy/stats/_multivariate.py]: Having this paragraph is better then not having it, but could you reword it a bit. Specifically, "vectorization" is heavily used for something different in numpy land (whether it's -s- or -z- :-). Would also be good to mention that this is equivalent but algorithmically inefficient.

**Comment by ev-br** [comment]
LGTM modulo a nitpick.
Extra commits would be good to squash (git rebase -I HEAD~4), but we could do it if you are not easy with it.

**Comment by argriffing** [comment]
Thanks @drpeteb!

BTW I checked out this branch locally to run `$ python runtests.py --doctests -s stats` but this has 94 test errors and 33 test failures and popped up some plot windows. I assume none of that is due to this PR, but rather that scipy docstring examples are not really interpretable as doctests.

**Comment by ev-br** [comment]
@argriffing you probably want `--refguide-check` not `--doctests`.

**Comment by argriffing** [comment]
Is `--doctests` still a thing? Should it be removed from `runtests.py`?

**Comment by ev-br** [comment]
My experience with --doctests matches what you wrote above, somaybe we're better off aliasing it to ---refguide-check ( which does run doctests with a modified checker).

**Comment by argriffing** [comment]
With `--refguide-check` I still get a handful of errors, mostly involving passing floats when integers are expected. I assume this is because I am using a newish numpy version that must not be tested with scipy on TravisCI? Also there is a SuperLU error.

**Comment by ev-br** [comment]
> I still get a handful of errors, 

Would be good to fix I guess (unless these are numpy 1.10.1 regressions which are going to be fixed in 1.10.2?)

**Comment by argriffing** [comment]
> maybe we're better off aliasing it to --refguide-check

I looked at this... maybe it would make sense if `refguide_check.py` gets an `--only-doctests` flag or something. Or maybe if it's renamed to `doc_check.py` with `--refguide`, `--doctests`, and `--tutorials` flags.

**Pull request merged**